java-squid/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -54,11 +54,12 @@
   private final ComplexityVisitor complexityVisitor;
   private final File file;
   private final Integer javaVersion;
+  private final boolean failedParsing;
   private final Map<Class<? extends SECheck>, SetMultimap<Tree, String>> seIssues = new HashMap<>();
 
   public DefaultJavaFileScannerContext(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)